### ISSUE TITLE
Dealing with empty masks.

### ISSUE THREAD

**Issue by BBQuercus** [mpicbg-csbd/stardist#25]
Occurred while training a 2d model. From a public dataset, I had the case, that one of the masks was empty (only background). It took a while to figure out what went wrong. I attached the error below. I am not sure where you might want to add the resolving code (as the error occurs in csbdeep) – therefore not a pull request. I came up with two potential solutions.

Preprocessing:
`X, Y = zip(*[(x, y) for x, y in zip(X, Y) if y.max() >= 1])`
Runtime:
`assert / raise y.max() >=1`.

```---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-25-50a4e0118c27> in <module>
      1 model.train(X_trn, Y_trn, validation_data=(X_val, Y_val), 
----> 2             augmenter=Augmenter(**data_gen_args).augment)

~/anaconda3/envs/image/lib/python3.7/site-packages/stardist/models/model2d.py in train(self, X, Y, validation_data, augmenter, seed, epochs, steps_per_epoch)
    352         Xv, Mv, Pv, Dv = [None]*n_take, [None]*n_take, [None]*n_take, [None]*n_take
    353         for i,k in enumerate(ids):
--> 354             (Xv[i],Mv[i]),(Pv[i],Dv[i]) = data_val[k]
    355         Xv, Mv, Pv, Dv = np.concatenate(Xv,axis=0), np.concatenate(Mv,axis=0), np.concatenate(Pv,axis=0), np.concatenate(Dv,axis=0)
    356         data_val = [[Xv,Mv],[Pv,Dv]]

~/anaconda3/envs/image/lib/python3.7/site-packages/stardist/models/model2d.py in __getitem__(self, i)
     49         arrays = [sample_patches_from_multiple_stacks((self.Y[k],) + self.channels_as_tuple(self.X[k]),
     50                                                       patch_size=self.patch_size, n_samples=1,
---> 51                                                       patch_filter=self.no_background_patches_cached(k)) for k in idx]
     52         if self.n_channel is None:
     53             X, Y = list(zip(*[(x[0][self.b],y[0]) for y,x in arrays]))

~/anaconda3/envs/image/lib/python3.7/site-packages/stardist/models/model2d.py in <listcomp>(.0)
     49         arrays = [sample_patches_from_multiple_stacks((self.Y[k],) + self.channels_as_tuple(self.X[k]),
     50                                                       patch_size=self.patch_size, n_samples=1,
---> 51                                                       patch_filter=self.no_background_patches_cached(k)) for k in idx]
     52         if self.n_channel is None:
     53             X, Y = list(zip(*[(x[0][self.b],y[0]) for y,x in arrays]))

~/anaconda3/envs/image/lib/python3.7/site-packages/csbdeep/data/generate.py in sample_patches_from_multiple_stacks(datas, patch_size, n_samples, datas_mask, patch_filter, verbose)
     98 
     99     if n_valid == 0:
--> 100         raise ValueError("'patch_filter' didn't return any region to sample from")
    101 
    102     sample_inds = choice(range(n_valid), n_samples, replace=(n_valid < n_samples))

ValueError: 'patch_filter' didn't return any region to sample from```

**Comment by maweigert** [comment]
Thanks for reporting this! Indeed that is a bug - we'll look into how to best fix it

**Comment by uschmidt83** [comment]
This problem should be fixed in release 0.4.0.
